File: BalmacedaAV/Laboratorio3Python/Leccion13/Gerente.py

from Empleado import Empleado

class Gerente(Empleado):
    def __init__(self, nombre, sueldo, departamento):
        super().__init__(nombre, sueldo)
        self.departamento = departamento

    def __str__(self):  #self: Gerente [ Departamento: Sistemas] Empleado: [Nombre: Noelia, Sueldo: 60000]
        return f'Gerente: [Departamento: {self.departamento}], Empleado: [{self.nombre}, Sueldo: {self.sueldo}]'

    #ME DA ERROR
    #    def __str__(self):
        #return f'Gerente [ Departamento: {self.departamento}] {super().__str__()}'

    def mostrar_detalles(self):
        return self.__str__()

File: EmiliaBazan/Java/Leccion04/Sobreescritura/src/domain/Gerente.java
/*
 * Sobreescritura
 */
package domain;

/*
 * @author Adriana
 */
public class Gerente extends Empleado{
    private String departamento;
    
    public Gerente(String nombre, double sueldo, String departamento){
        super(nombre, sueldo);
        this.departamento = departamento;
    }
    
    //Sobreescribimos el método
    @Override
    //public String obtenerDetalles(){ //para testeos
    public String obtenerDetalles(){
        return super.obtenerDetalles()+", Departamento: "+this.departamento;
    }
}
